src/azul/openapi/validation.py

```diff
@@ -0,0 +1,149 @@
+import json
+import logging
+from typing import (
+    Mapping,
+    Optional,
+)
+
+from chalice import (
+    BadRequestError,
+)
+import jsonschema
+from more_itertools import (
+    one,
+)
+
+from azul.openapi.schema import (
+    type_lookup,
+)
+from azul.uuids import (
+    InvalidUUIDError,
+    InvalidUUIDVersionError,
+    validate_uuid,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class SpecValidator:
+
+    def __init__(self,
+                 *,
+                 name: str,
+                 required: bool,
+                 content: Optional[Mapping] = None,
+                 pattern: Optional[str] = None,
+                 schema: Optional[Mapping] = None,
+                 **kwargs):
+        """
+        :param kwargs: Since 'in' is a Python reserved word we can only extract it
+         using kwargs. All other kwargs are ignored.
+        """
+        self.content = content
+        self.pattern = pattern
+        self.name = name
+        self.required = required
+        self.schema = schema
+        self.json_validators = dict(jsonschema.Draft4Validator.VALIDATORS)
+        self._in = kwargs['in']
+        if self.schema is not None and self.content is not None:
+            raise TypeError('OpenAPI parameter specification enforces that schema'
+                            ' and content are mutually exclusive')
+        if self.schema:
+            self.validator = type_lookup(self.schema['type'])
+        elif self.content:
+            self.validator = type_lookup(self.content['application/json']['schema']['type'])
+        else:
+            raise TypeError('Parameters must contain either a schema property or'
+                            ' content property')
```

Comment:
A slightly more succinct way to write this would be:
```suggestion
        if self.schema is not None and self.content is None:
            self.validator = type_lookup(self.schema['type'])
        elif self.content is not None and self.schema is None:
            self.validator = type_lookup(self.content['application/json']['schema']['type'])
        else:
            raise TypeError('Parameters must contain either schema or content'
                            '(but not both)')
```